tst/unit/test_pararrays.cpp

```diff
@@ -0,0 +1,426 @@
+//========================================================================================
+// Parthenon performance portable AMR framework
+// Copyright(C) 2020 The Parthenon collaboration
+// Licensed under the 3-clause BSD License, see LICENSE file for details
+//========================================================================================
+// (C) (or copyright) 2020. Triad National Security, LLC. All rights reserved.
+//
+// This program was produced under U.S. Government contract 89233218CNA000001
+// for Los Alamos National Laboratory (LANL), which is operated by Triad
+// National Security, LLC for the U.S. Department of Energy/National Nuclear
+// Security Administration. All rights in the program are reserved by Triad
+// National Security, LLC, and the U.S. Department of Energy/National Nuclear
+// Security Administration. The Government is granted for itself and others
+// acting on its behalf a nonexclusive, paid-up, irrevocable worldwide license
+// in this material to reproduce, prepare derivative works, distribute copies to
+// the public, perform publicly and display publicly, and to permit others to do
+// so.
+//========================================================================================
+
+// C++ headers
+#include <cmath>
+#include <iostream>
+#include <random>
+#include <string>
+
+// 3rd-party headers
+#include <catch2/catch.hpp>
+
+// Parthenon headers
+#include "kokkos_abstraction.hpp"
+#include "parthenon_arrays.hpp"
+
+using parthenon::DevSpace;
+using parthenon::ParArrayND;
+using parthenon::ParArray3D;
+using Real = double;
+
+constexpr int N = 32 + 2;
+constexpr int NT = 100;
+constexpr int NARRAYS = 64;
+
+#ifdef KOKKOS_ENABLE_CUDA
+using UVMSpace = Kokkos::CudaUVMSpace;
+#else // all on host
+using UVMSpace = DevSpace;
+#endif
+
+KOKKOS_INLINE_FUNCTION Real coord(const int i, const int n) {
+  const Real dx = 2.0/(n+1.0);
+  return -1.0 + dx*i;
+}
+
+KOKKOS_INLINE_FUNCTION Real gaussian(const int iz, const int iy, const int ix,
+                                     const int nz, const int ny, const int nx) {
+  Real x = coord(ix,nx);
+  Real y = coord(iy,ny);
+  Real z = coord(iz,nz);
+  Real r2 = x*x + y*y + z*z;
+  return exp(-r2);
+}
+
+KOKKOS_INLINE_FUNCTION Real gaussian(const int iz, const int iy, const int ix) {
+  return gaussian(iz,iy,ix,N,N,N);
+}
+
+template<typename T>
+KOKKOS_FORCEINLINE_FUNCTION
+void stencil(T& l, T& r, const int k, const int j, const int i) {
+  l(k,j,i) = (1./6.)*( r(k-1, j,   i)   + r(k+1, j,   i)
+                      +r(k,   j-1, i)   + r(k,   j+1, i)
+                      +r(k,   j,   i-1) + r(k,   j,   i+1));
+}
+
```

Comment:
If you care about the formatting you might want to add the // clang-format off // clang-format on blocks.

Reply:
Good catch.